lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
+        return cb(err);
+      }
```

Comment:
@kjdelisle hmm, I don't get this idea: "but if we want to preserve the idea of what updateOrCreate does from a LoopBack perspective, then we should use updateAttributes to attempt the update", from my understanding, from a BL perspective it's the way inverse, `updateAttributes` should call `updateAll`(alias for update).
And if updateOrCreate calls other method, it should be `update` but not `updateAttributes`